File: lab/python/textos/cpf_valido_com_regex.py
import re


def is_cpf(cpf):
    cpf = cpf.strip()
    return bool(re.match(r"^\d{3}\.\d{3}\.\d{3}-\d{2}$", cpf))


cpfs = [
    "123.456.789-10",
    "111.111.111-11",
    "12345678910",
    "12.3456.789-10",
    "123.456.789-100"
]

for cpf in cpfs:
    print(f"{cpf} -> {'Válido' if is_cpf(cpf) else 'Inválido'}")

File: lab/javascript/textos/cpf_valido_com_regex.js
function is_cpf(cpf) {
  cpf = cpf.trim();
  const regex = /^\d{3}\.\d{3}\.\d{3}-\d{2}$/;
  return regex.test(cpf);
}

const cpfs = [
  "123.456.789-10",
  "111.111.111-11",
  "12345678910",
  "12.3456.789-10",
  "123.456.789-100"
];

for (const cpf of cpfs) {
  console.log(`${cpf} -> ${is_cpf(cpf) ? "Válido" : "Inválido"}`);
}
